src/amo/components/Search/index.js

```diff
@@ -40,12 +40,29 @@ export class SearchBase extends React.Component {
     count: 0,
     enableSearchSort: true,
     filters: {},
+    paginationQueryParams: null,
     pathname: '/search/',
     results: [],
   }
 
   componentWillMount() {
-    const { dispatch, filters } = this.props;
+    this.dispatchSearch({ filters: this.props.filters });
+  }
+
+  componentWillReceiveProps({ filters: newFilters }) {
+    const { filters: oldFilters } = this.props;
+
+    if (!deepEqual(oldFilters, newFilters)) {
+      this.dispatchSearch({ filters: newFilters });
+    }
+  }
+
+  dispatchSearch({ filters = {} } = {}) {
+    const { dispatch } = this.props;
+
+    if (hasSearchFilters(filters)) {
```

Comment:
ok, then maybe you can combine this if statement with the `!deepEqual(...)` one above? It would be nice to see the conditional logic all in one place (that was the source of my confusion). As mentioned in previous comments, I think you will need to do the `!deepEqual(...)` every time anyway, e.g. to handle the server side rendering case, so you might as well combine them.